pkg/appfile/parser.go

```diff
@@ -406,3 +435,93 @@ func PrepareProcessContext(k8sClient client.Client, wl *Workload, applicationNam
 	}
 	return pCtx, nil
 }
+
+// GetOutputSecretNames set all secret names, which are generated by cloud resource, to context
+func GetOutputSecretNames(workloads *Workload) (string, error) {
+	secretName, err := getComponentSetting(process.OutputSecretName, workloads.Params)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprint(secretName), nil
+}
+
+func parseWorkloadK8sSecretSchema(ctx context.Context, c client.Client, namespace string, wl *Workload) ([]process.RequiredSecrets, error) {
+	var requiredSecret []process.RequiredSecrets
+	api, err := utils.GenerateOpenAPISchemaFromDefinition(wl.Name, wl.Template)
+	if err != nil {
+		if !errors.Is(err, errors.Errorf(utils.ErrNoSectionParameterInCue, wl.Name)) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	schema, err := utils.ConvertOpenAPISchema2SwaggerObject(api)
+	if err != nil {
+		return nil, err
+	}
+	tag := utils.K8sSecretSchemaTag
```

Comment:
why use a variable instead of directly use `utils.K8sSecretSchemaTag` ?

Reply:
Use `utils.K8sSecretSchemaTag` will be more clean code to read.

Reply:
Because `utils.K8sSecretSchemaTag` it used twice in this function.

Reply:
tag is also used twice? they are all single var

Reply:
Done.